src/ServiceControl/DbMigrations/1.27/SplitFailedMessageDocumentsMigration.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace Particular.ServiceControl.DbMigrations
+{
+    using System.Linq;
+    using global::ServiceControl;
+    using global::ServiceControl.MessageFailures;
+    using global::ServiceControl.Operations;
+    using global::ServiceControl.Recoverability;
+    using NServiceBus.ObjectBuilder;
+    using Raven.Client;
+
+    public class SplitFailedMessageDocumentsMigration : IMigration
+    {
+        public SplitFailedMessageDocumentsMigration(IBuilder builder)
+        {
+            var failedEnrichers = builder.BuildAll<IFailedMessageEnricher>().ToArray();
+
+            failedMessageFactory = new FailedMessageFactory(failedEnrichers);
+        }
+
+        public void Apply(IDocumentSession session)
+        {
+            var currentPage = 0;
+            int retrievedResults;
+
+            do
+            {
+                var failedMessages = session.Advanced.LoadStartingWith<FailedMessage>(
+                    FailedMessage.MakeDocumentId(string.Empty),
+                    start: PageSize*currentPage,
+                    pageSize: PageSize);
+
+                currentPage++;
+
+                retrievedResults = failedMessages.Length;
+
+                foreach (var failedMessage in failedMessages)
+                {
+                    Check(failedMessage, session);
+                }
+
+            } while (retrievedResults == PageSize);
+        }
+
+        private void Check(FailedMessage failedMessage, IDocumentSession session)
+        {
+            var originalRecords = failedMessage.ProcessingAttempts.Select(x => new ProcessingAttemptRecord(x)).ToArray();
```

Comment:
What if we have a FailedMessage that we retried but it has one of these duplicates in it? Shouldn't we pull the duplicate out?